tools/ci/actions/check-parsetree-modified.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Hygiene Checks: check that whenever the parsetree.mli file has been modified,
+# the parsetree-change label has been applied to the PR
+
+# Exactly of the following must be true:
+#   - No commit in the PR alters the parsetree.mli file
+#   - The parsetree-change label is applied to the PR
+
+API_URL="$1"
+shift 1
+
+AUTH="authorization: Bearer $GITHUB_TOKEN"
+
+# We need all the commits in the PR to be available
+. tools/ci/actions/deepen-fetch.sh
+
+COMMIT_RANGE="$MERGE_BASE..$PR_HEAD"
+
+LABEL='parsetree-change'
+
+if git diff "$COMMIT_RANGE" --name-only --exit-code parsing/parsetree.mli \
```

Comment:
You are missing a `!` negation here, `git diff` fails if there were no difference.